source/dub/compilers/dmd.d

```diff
@@ -54,6 +55,62 @@ class DMDCompiler : Compiler {
 
 	@property string name() const { return "dmd"; }
 
+	string version_(in ref BuildPlatform platform)
+	{
+		import std.regex : matchFirst, ctRegex;
+		import std.string : lineSplitter;
+
+		auto r = ctRegex!(`v(\d+\.\d+\.\d+(-\w+)?)$`, "m");
+		string ver;
+
+		invokeTool([ platform.compilerBinary, "--version" ], (int code, string output) {
+			enforce(code == 0, "Could not run `"~platform.compilerBinary~" --version`");
+			auto c = matchFirst(output, r);
+			enforce(c, "Could not parse DMD version");
+			assert(c.length >= 2);
+			c.popFront();
+			ver = c.front;
+		});
```

Comment:
This is yet another opened process. It would be great to combine this with the platform probing which already happens anyways.
Though the platform probing isn't ideal either and we wanted to replace it with DMD's Json interface for quite a while...

Reply:
`std.compiler` can probably help, but I don't think I can get the version micro number with it.